File: day7/day7.ex
testInput = String.split("pbga (66)
xhth (57)
ebii (61)
havc (66)
ktlj (57)
fwft (72) -> ktlj, cntj, xhth
qoyq (66)
padx (45) -> pbga, havc, qoyq
tknk (41) -> ugml, padx, fwft
jptl (61)
ugml (68) -> gyxo, ebii, jptl
gyxo (61)
cntj (57)", "\n")

defmodule Day7 do
  def findParents(list) do
    childless = Enum.filter(list, &(length(String.split(&1, " -> ")) == 1))
    clean = Enum.map(childless, &(cleanNode(&1)))
    sorted = Enum.filter(list, &(length(String.split(&1, " -> ")) > 1))
    completed = listToMap(sorted)
    joinTowers(completed)
  end

  def listToMap(list) do
    Enum.map(list, &(String.split(&1, " -> ")))
    |> Map.new(fn[a, b] -> {cleanNode(a), String.split(b, ", ")} end)
  end

  def cleanNode(node) do
    Enum.at(String.split(node), 0)
  end

  def buildTower(cl, co) do
    if length(cl) > 0 do

    else
      co
    end
  end

  def joinTowers(co, final\\ %{}) do
    if length(Map.keys(co)) > 1 do
      Enum.map(co, fn(t) -> {IO.inspect(t)} end)
    else
      co
    end
  end
end

IO.inspect(Day7.findParents(testInput))

File: day7/day7.js
const testInput = "pbga (66)\n" +
"xhth (57)\n" +
"ebii (61)\n" +
"havc (66)\n" +
"ktlj (57)\n" +
"fwft (72) -> ktlj, cntj, xhth\n" +
"qoyq (66)\n" +
"padx (45) -> pbga, havc, qoyq\n" +
"tknk (41) -> ugml, padx, fwft\n" +
"jptl (61)\n" +
"ugml (68) -> gyxo, ebii, jptl\n" +
"gyxo (61)\n" +
"cntj (57)";

const findParent = (arr) => {
  const childless = [];
  const completed = {};
  arr.map(item => {
    if (item.split('->').length > 1) {
      completed[item.split('->')[0].trim()] = item.split('->')[1].trim();
    }
    else {
      childless.push(item)
    }
  });
  return completed;
};

const buildTower = (cl, co) => {
  if (cl.length > 1 && Object.keys(co).length > 1) {

  }
  else {
    return co;
  }
};

console.log(findParent(testInput.split('\n')));
